File: trunk/JAVA/jzbus/src/test/java/net/zbus/asyn/ClientSend.java
package net.zbus.asyn;

import net.zbus.AsynCtrl;
import net.zbus.BusClient;
import net.zbus.ConnectionConfig;
import net.zbus.ZMsg;

public class ClientSend {
 
	public static void main(String[] args) { 
		ConnectionConfig config = new ConnectionConfig();
		BusClient client = new BusClient(config);
		AsynCtrl ctrl = new AsynCtrl();
		ctrl.setService("helloworld");
		ctrl.setMessageId("10001");
		ctrl.setPeerId("local_mq"); 
		
		while(true){
			try{
				ZMsg msg = new ZMsg();
				msg.pushBack("asyn call hello");
				ZMsg res = client.send(ctrl, msg);
				res.dump();
				break;
			}catch (Exception e) {
				break;
			}
		}
		
		client.destroy();
	} 
}

File: trunk/MS.NET/clrzbus/examples/asyn/ClientSend.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class ClientSend
    {
        public static void Main(string[] args)
        {

            ConnectionConfig config = new ConnectionConfig();
            config.Host = "127.0.0.1";
            config.Port = 15555;
            //config.Verbose = true;

            BusClient client = new BusClient(config);

            AsynCtrl ctrl = new AsynCtrl();
            ctrl.Service = "helloworld";
            ctrl.PeerId = "local_mq";
            ctrl.MessageId = "10000";

            ZMsg msg = new ZMsg();
            msg.PushBack("C# asyn call");
            msg = client.Send(ctrl, msg);
            msg.Dump();

            client.Destroy();

            Console.ReadKey();
        } 
    }
}
